cdap-watchdog/src/main/java/co/cask/cdap/metrics/data/DefaultMetricsTableFactory.java

```diff
@@ -58,44 +56,45 @@
   @Inject
   public DefaultMetricsTableFactory(final CConfiguration cConf,
                                     final DatasetFramework dsFramework) {
-    try {
-      this.cConf = cConf;
-      this.dsFramework =
-        new NamespacedDatasetFramework(dsFramework,
-                                       new DefaultDatasetNamespace(cConf, Namespace.SYSTEM));
-
-      this.entityCodecs = CacheBuilder.newBuilder().build(new CacheLoader<String, MetricsEntityCodec>() {
-        @Override
-        public MetricsEntityCodec load(String namespace) throws Exception {
-          String tableName = namespace.toLowerCase() + "." + cConf.get(MetricsConstants.ConfigKeys.ENTITY_TABLE_NAME,
-                                                                       MetricsConstants.DEFAULT_ENTITY_TABLE_NAME);
-          MetricsTable table = getOrCreateMetricsTable(tableName, DatasetProperties.EMPTY);
-          EntityTable entityTable = new EntityTable(table);
-
-          return new MetricsEntityCodec(entityTable,
-                                        MetricsConstants.DEFAULT_CONTEXT_DEPTH,
-                                        MetricsConstants.DEFAULT_METRIC_DEPTH,
-                                        MetricsConstants.DEFAULT_TAG_DEPTH);
+    this.cConf = cConf;
+    this.dsFramework =
+      new NamespacedDatasetFramework(dsFramework,
+                                     new DefaultDatasetNamespace(cConf, Namespace.SYSTEM));
+
+    this.entityCodec = Suppliers.memoize(new Supplier<MetricsEntityCodec>() {
+
+      @Override
+      public MetricsEntityCodec get() {
+        String tableName = cConf.get(MetricsConstants.ConfigKeys.ENTITY_TABLE_NAME,
+                                     MetricsConstants.DEFAULT_ENTITY_TABLE_NAME);
+        MetricsTable table;
+        try {
+          table = getOrCreateMetricsTable(tableName, DatasetProperties.EMPTY);
+        } catch (Exception e) {
+          throw Throwables.propagate(e);
         }
-      });
-    } catch (Exception e) {
-      throw Throwables.propagate(e);
-    }
+        EntityTable entityTable = new EntityTable(table);
+
+        return new MetricsEntityCodec(entityTable,
+                                      MetricsConstants.DEFAULT_CONTEXT_DEPTH,
+                                      MetricsConstants.DEFAULT_METRIC_DEPTH,
+                                      MetricsConstants.DEFAULT_TAG_DEPTH);
+      }
+    });
   }
 
   @Override
-  public TimeSeriesTable createTimeSeries(String namespace, int resolution) {
```

Comment:
nono - that was really misleading. That was a "scope" (that is removed in this PR) - but param was called namespace... The idea was for each scope create different tables for metrics. Now we use single set of those.